modules/tinymce/src/core/main/ts/api/EditorOptions.ts

```diff
@@ -0,0 +1,284 @@
+
+import Editor from './Editor';
+import { EditorOptions, NormalizedEditorOptions } from './OptionTypes';
+
+interface ProcessorSuccess<T> {
+  valid: true;
+  value: T;
+}
+
+interface ProcessorError {
+  valid: false;
+  message: string;
+}
+
+type SimpleProcessor = (value: unknown) => boolean;
+type Processor<T> = (value: unknown) => ProcessorSuccess<T> | ProcessorError;
+
+export interface BuiltInOptionTypeMap {
+  'string': string;
+  'number': number;
+  'boolean': boolean;
+  'string[]': string[];
+  'array': any[];
+  'function': Function;
+  'object': any;
+}
+
+export type BuiltInOptionType = keyof BuiltInOptionTypeMap;
+
+interface BaseOptionSpec {
+  immutable?: boolean;
+  deprecated?: boolean;
+  docsUrl?: string;
+}
+
+export interface BuiltInOptionSpec<K extends BuiltInOptionType> extends BaseOptionSpec {
+  processor: K;
+  default?: BuiltInOptionTypeMap[K];
+}
+
+export interface SimpleOptionSpec<T> extends BaseOptionSpec {
+  processor: SimpleProcessor;
+  default?: T;
+}
+
+export interface OptionSpec<T, U> extends BaseOptionSpec {
+  processor: Processor<U>;
+  default?: T;
+}
+
+/**
+ * TinyMCE Options API.
+ *
+ * @class tinymce.editor.Options
+ */
+
+export interface Options {
+  /**
+   * Register a new option that can be used within TinyMCE.
+   *
+   * @method register
+   * @param {String} name Name of an option.
+   * @param {OptionSpec} spec An option spec describing how to validate the option and other optional metadata.
+   */
+  register: {
+    <K extends BuiltInOptionType>(name: string, spec: BuiltInOptionSpec<K>): void;
+    <K extends keyof NormalizedEditorOptions>(name: K, spec: OptionSpec<NormalizedEditorOptions[K], EditorOptions[K]> | SimpleOptionSpec<NormalizedEditorOptions[K]>): void;
+    <T, U>(name: string, spec: OptionSpec<T, U>): void;
+    <T>(name: string, spec: SimpleOptionSpec<T>): void;
+  };
+
+  /**
+   * Check if an option has been registered.
+   *
+   * @method isRegistered
+   * @param {String} name Name of an option.
+   * @return {Boolean} True if the option has already been registered, otherwise false.
+   */
+  isRegistered: (name: string) => boolean;
+
+  /**
+   * Get the value of a registered option.
+   *
+   * @method get
+   * @param {String} name Name of a registered option.
+   * @return {Object} An option value or the registered default if not set.
+   */
+  get: {
+    <K extends keyof EditorOptions>(name: K): EditorOptions[K] | undefined;
+    <T>(name: string): T | undefined;
+  };
+
+  /**
+   * Set the value for a registered option.
+   *
+   * @method set
+   * @param {String} name Name of a registered option.
+   * @return {Boolean} True if the option value was successfully set, otherwise false.
+   */
+  set: {
+    <K extends keyof EditorOptions>(name: K, value: EditorOptions[K]): boolean;
+    <T>(name: string, value: T): boolean;
+  };
+
+  /**
+   * Clear any set value for the specified option so that it'll revert back to using the default value.
```

Comment:
```suggestion
   * Clear the set value for the specified option and revert back to the default value.
```
`the` to be specific?